rust_src/src/keymap.rs

```diff
@@ -16,6 +16,45 @@ pub fn Ctl(c: char) -> i32 {
     (c as i32) & 0x1f
 }
 
+/// Construct and return a new keymap, of the form (keymap CHARTABLE . ALIST).
+/// CHARTABLE is a char-table that holds the bindings for all characters
+/// without modifiers.  All entries in it are initially nil, meaning
+/// "command undefined".  ALIST is an assoc-list which holds bindings for
+/// function keys, mouse events, and any other things that appear in the
+/// input stream.  Initially, ALIST is nil.
+///
+/// The optional arg STRING supplies a menu name for the keymap
+/// in case you use it as a menu with `x-popup-menu'.
+#[lisp_fn(min = "0")]
+pub fn make_keymap(string: LispObject) -> LispObject {
+    let tail: LispObject;
+    if !string.is_nil() {
+        tail = LispObject::from_raw(unsafe { list1(string.to_raw()) });
+    } else {
+        tail = LispObject::constant_nil();
+    }
+
+    let char_table = unsafe { Fmake_char_table(Qkeymap, LispObject::constant_nil().to_raw()) };
```

Comment:
Instead of using `to_raw` on `constant_nil` just use `Qnil` which you can import from remacs-sys.

Reply:
I actually did this the first time but figured that less imports are better. Will do it this way going forward.